h2/src/main/org/h2/command/Parser.java

```diff
@@ -4861,7 +4861,7 @@ private String readIdentifierWithSchema2(String s) {
         schemaName = s;
         s = readColumnIdentifier();
         if (currentTokenType == DOT) {
-            if (equalsToken(schemaName, database.getShortName())) {
+            if (equalsToken(schemaName, database.getShortName()) || database.getIgnoreCatalogs()) {
```

Comment:
`Parser.parseComment()` needs similar logic.

Reply:
The case is, I create tests for existing modules, developed by other teams. So changing of their code is the very last option. They use catalogs in their productive code, my problem is how to use H2 in more complicated tests. Usage of multiple h2-databases is also no option since I must be able to handle the accessing of different catalogs in one transaction.

Reply:
I will look into parsecomment, and handle the test-error.

Reply:
See TestIgnoreCatalogs.canCommentOn()